web/modules/custom/pineapple_general/pineapple_general.module

```diff
@@ -108,6 +93,56 @@ function pineapple_general_form_alter(&$form, \Drupal\Core\Form\FormStateInterfa
   }
 }
 
+/**
+ * Implements hook_field_widget_WIDGET_TYPE_form_alter().
+ */
+function pineapple_general_field_widget_entity_reference_paragraphs_form_alter(&$element, \Drupal\Core\Form\FormStateInterface $form_state, $context) {
+  $user_roles = \Drupal::currentUser()->getRoles();
+   /* For each item in the widget array, if paragraph type value is equal to
+   image_paragraph. */
+ if (isset($element['subform']['field_p_c_component']) && !in_array('administrator', $user_roles)) {
+   foreach ($element['subform']['field_p_c_component']['widget'] as &$paragraph) {
+     if (isset($paragraph['#paragraph_type'])) {
+       if ($paragraph['#paragraph_type'] == 'image_paragraph' || $paragraph['#paragraph_type'] == 'image_grid') {
+         if (isset($paragraph['subform']['field_p_ip_hero_image']) || isset($paragraph['subform']['field_p_ig_item'])) {
+           /* For each $parent in the field_parents array, if the parent is in an
+           article or recipe content type, restrict access to the hero image field. */
```

Comment:
Review this comment and the one on line 114 please

Reply:
Done